### PR TITLE
Portal layout and content fixes

### PR DESCRIPTION
**Pull request by sandratoh**
- New documents needed section in steps that require document upload or info
- Parcel lookup fields update + relevant step 8 fields:
  - Remove "PIN" from Fee Simple UI (also in ALCS)
  - Remove "Parcel Purchase Date" from Crown Land UI (also in ALCS)
  - Reset parcel fields when switching parcel ownership type
  - New modal to confirm parcel ownership type switching if parcel fields are dirty
- Clean up "Other Parcels in the Community" UI and parcel labels
- Match toggle buttons and upload section error outlines with input fields (missed in previous ticket)
- Create new parcel ownership type enum

### CODE REVIEW

File: portal-frontend/src/app/features/application-details/parcel/parcel.component.html [hunk 1]
```diff
@@ -72,25 +74,34 @@ <h4>
...
-    <div class="grid-double">
-      {{ parcel.purchasedDate | date }}
-      <app-no-data
-        [showRequired]="showErrors && !!parcel.validation?.isPurchasedDateRequired"
-        *ngIf="!parcel.purchasedDate"
-      >
-      </app-no-data>
-    </div>
+    <!--  Remove field if fee simple selected  -->
+    <ng-container *ngIf="parcel.ownershipType?.code === 'CRWN'">
```
Comment (dhaselhan) [inline portal-frontend/src/app/features/application-details/parcel/parcel.component.html:78]: I'm not sure just hiding the field will be sufficient, if a user selects crown land then fills a pin then hits save, then toggles to fee simple the PIN would stay saved and still be present in the view page, PDF, ALCS, etc. You either need to add this ngIf to all of the above or clear it from the DB.

**Review by sandratoh — COMMENTED** [review]
Some questions re: code improvements

File: portal-frontend/src/app/features/application-details/parcel/parcel.component.html [hunk 2]
```diff
@@ -62,8 +62,10 @@ <h4>
         *ngIf="parcel.validation?.isMapAreaHectaresRequired || false"
       ></app-no-data>
     </div>
+    <!-- TODO: Create ownership type code enum -->
     <div class="subheading2 grid-1">
-      PID<app-updated-banner *ngIf="updatedFields.includes(parcelInd + '.pid')"></app-updated-banner>
+      PID {{ parcel.ownershipType?.code === 'CRWN' ? '(optional)' : ''
+      }}<app-updated-banner *ngIf="updatedFields.includes(parcelInd + '.pid')"></app-updated-banner>
```
[the remark this answers is not in the document]
Comment (dhaselhan) [reply portal-frontend/src/app/features/application-details/parcel/parcel.component.html:68]: Either is fine with me, if you want to create an enum you can put it in the DTO.

File: portal-frontend/src/app/features/edit-submission/parcel-details/parcel-details.component.html [hunk 3]
```diff
@@ -2,7 +2,14 @@
...
     <p>Provide parcel identification and registered ownership information for each parcel under application.</p>
-    <p>All fields are required unless stated optional.</p>
+    <p>*All fields are required unless stated optional.</p>
+    <div class="step-documents">
+      <h6>Documents needed for this step:</h6>
+      <ul>
+        <li>Certificate of Title</li>
+        <li>Corporate Summary (if applicable)</li>
+      </ul>
+    </div>
```
[the remark this answers is not in the document]
Comment (dhaselhan) [reply portal-frontend/src/app/features/edit-submission/parcel-details/parcel-details.component.html:12]: You can move the css class to the edit-submission components :host::ng-deep section, it has a few styles that are re-used among the steps. This means you can re-use the CSS class without needing to create a full component.

File: portal-frontend/src/app/features/edit-submission/parcel-details/parcel-entry/parcel-entry.component.ts [hunk 4]
```diff
@@ -326,7 +370,7 @@ export class ParcelEntryComponent implements OnInit {
       isConfirmedByApplicant: this.enableUserSignOff ? this.parcel.isConfirmedByApplicant : false,
     });
 
-    this.isCrownLand = this.parcelType.getRawValue() === 'CRWN';
+    this.isCrownLand = this.parcelType.value ? this.parcelType.getRawValue() === 'CRWN' : null;
```
Comment (dhaselhan) [inline portal-frontend/src/app/features/edit-submission/parcel-details/parcel-entry/parcel-entry.component.ts:373]: Use enum for CRWN

File: portal-frontend/src/app/features/edit-submission/parcel-details/parcel-entry/parcel-entry.component.ts [hunk 5]
```diff
@@ -172,28 +180,64 @@ export class ParcelEntryComponent implements OnInit {
...
+      return changeParcelType();
     }
-
-    this.updateParcelOwners([]);
-    this.filteredOwners = this.mapOwners(this.owners);
-    this.pid.updateValueAndValidity();
   }
 
-  async attachFile(file: FileHandle, documentType: DOCUMENT_TYPE, parcelUuid: string) {
+  async attachFile(file: FileHandle, _documentType: DOCUMENT_TYPE, parcelUuid: string) {
```
Comment (dhaselhan) [inline portal-frontend/src/app/features/edit-submission/parcel-details/parcel-entry/parcel-entry.component.ts:240]: If this parameter isnt used, should it be deleted?

**Review by dhaselhan — APPROVED** [review]
LGTM 👍

File: portal-frontend/src/app/features/edit-submission/parcel-details/parcel-entry/parcel-entry.component.ts (hunk 5 above)
Comment (sandratoh) [reply portal-frontend/src/app/features/edit-submission/parcel-details/parcel-entry/parcel-entry.component.ts:240]: Looks like we can remove it. The document type is specified in the service, we probably just forgot to update here. I'll delete it here!

**Pull request merged**